### PR TITLE
submit interfaces to encode and decode binlog

### PR DESCRIPTION
**Pull request by chaychen2005**

### CODE REVIEW

File: libstorage/BinLogHandler.cpp [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+
+BinLogHandler::~BinLogHandler()
+{
+    if (m_outBinaryFile.is_open())
+    {
+        m_outBinaryFile.close();
+    }
+}
+
+void BinLogHandler::setBinLogStoragePath(const std::string& path)
```
Comment (bxq2011hust) [inline libstorage/BinLogHandler.cpp]: why binlog need path? what about use a fixed path?

Comment (chaychen2005) [reply libstorage/BinLogHandler.cpp]: Done. Directory splicing is implemented externally.

File: libstorage/BinLogHandler.h [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+{
+public:
+    BinLogHandler();
+    virtual ~BinLogHandler();
+    void setBinLogStoragePath(const std::string& path);
+    bool writeBinLog(int64_t num, const std::vector<TableData::Ptr>& datas);
+    bool scanBinLog();
+
+private:
+    void writeUINT32(bytes&, uint32_t) {}
```
[the remark this answers is not in the document]
Comment (chaychen2005) [reply libstorage/BinLogHandler.h]: already done

File: libstorage/BinLogHandler.cpp [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+    m_path = path + "/binlog/";
+    fs::create_directory(m_path);
+}
+
+bool BinLogHandler::writeBinLog(int64_t num, const std::vector<TableData::Ptr>& datas)
+{
+    // encode
+    bytes buffer;
+    encodeBlock(num, datas, buffer);
+    // write binary file
```
[the remark this answers is not in the document]
Comment (chaychen2005) [reply libstorage/BinLogHandler.cpp]: already done

**Pull request merged**